src/bootstrap/src/core/build_steps/dist.rs

```diff
@@ -2599,3 +2599,38 @@ impl Step for Gcc {
         tarball.generate()
     }
 }
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct Enzyme {
+    pub target: TargetSelection,
+}
+
+impl Step for Enzyme {
+    type Output = GeneratedTarball;
+    const DEFAULT: bool = false;
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.alias("enzyme")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(Enzyme { target: run.target });
+    }
+
+    fn run(self, builder: &Builder<'_>) -> Self::Output {
+        let mut tarball = Tarball::new(builder, "enzyme", &self.target.triple);
+        let enzyme = builder.ensure(super::llvm::Enzyme { target: self.target });
+        tarball.set_overlay(OverlayKind::Enzyme);
+        tarball.is_preview(true);
+        if let Some(llvm_config) = builder.llvm_config(builder.config.build) {
+            let llvm_version_major = llvm::get_llvm_version_major(builder, &llvm_config);
+            let lib_ext = std::env::consts::DLL_EXTENSION;
```

Comment:
This won't work for cross-compilation, if someone wanted to venture there =D I guess that we could just enumerate all files in the `libenzyme` directory and check if they are dylibs (`helpers::is_dylib`) and have the required prefix (`libEnzyme`) and copy those.

Reply:
I'm afraid I'd first need to fix apple support (or windows) before we can do apple cross-compilation testing: https://github.com/rust-lang/rust/issues/140137
In theory we could compile from aarch linux to x86-64 linux (or vice versa), but not sure how interesting that is.
Shall I create an issue for it?